### PR TITLE
Add login functionality

### LINKED ISSUE #3: Implement login functionality
**Is your feature request related to a problem? Please describe.**
The login page UI is already completed. Now we need to make it functional by making a request to the backend and getting an access token. This token will be used further to make further backend API calls.

**Describe the solution you'd like**
- Once the user clicks on login after typing phone and password make an API call to the backend using this API- https://api.glific.com/#4a9a72a5-01d3-4af4-a7e0-7b50a67c6dac
- This will fetch an access token and it should move to a new page after that. Add a heading "Chat" on the new page

**Additional context**
Backend testing URL: https://api.staging.tides.coloredcow.com/api
Phone: 917834811114
Password: secret1234

Or you can setup the backend in local for ease in testing. Steps for setting up the backend :
https://github.com/glific/glific#readme

### PR DESCRIPTION
**Pull request by ajaman190**
## Description
This pull request implements the login functionality for the app. The login page UI was already completed and now this pull request adds the functionality to the page. 

## Changes Made
* Made an API call to the backend using the provided API
* Implemented a login function that fetches an access token
* Added a new page titled "Chat" that the user is navigated to after a successful login

## How to Test
To test the login functionality:
1. Open the app
2. Navigate to the login page
3. Enter the phone and password provided in the issue
4. Click "Continue"
5. If the login is successful, the user should be navigated to the "Chat" page.

## Screenshots
image

https://github.com/glific/mobile/assets/84832565/c0189812-7718-44c9-b3ca-a9f9be07abb9

## Related Issues
Closes #3

### CODE REVIEW

File: constants/API.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import axios, { AxiosInstance } from 'axios';
+
+function createAxiosClient(): AxiosInstance {
+  return axios.create({
+    baseURL: 'https://api.staging.tides.coloredcow.com/api',
```
Comment (mdshamoon) [inline constants/API.ts:5]: Can you keep this as an empty string for now. We will add an env file and pick it from there later.

File: constants/API.ts
Comment (mdshamoon) [inline constants/API.ts:1]: Change the directory name to config and file name to axios.ts

**Review by mdshamoon — CHANGES_REQUESTED** [review]
Looks good overall. Added some comments, please look into those.

**Comment by ajaman190** [comment]
I have made the requested changes, please review them.

**Pull request merged**